### DISCUSSION TITLE
想问一下，配置文件里配置微信证书的时候，推荐用绝对路径的原因

### DISCUSSION THREAD

**Discussion by rare0619** [w7corp/easywechat#2770] [Q&A] [answered]
项目用的hyperf，  会打包发布，在使用微信支付的平台证书的时候，看到文档注释会说平台证书需要使用绝对路径，但是使用绝对路径会造成需要硬编码路径前缀，感觉不是一个好方案，但是测试的时候使用相对路径读取证书也没有碰到问题，所以想问一下 使用相对路径读取证书会有什么潜在的风险？
![image](https://github.com/w7corp/easywechat/assets/11610757/072b3504-7f19-4115-bc2f-bbce188589bc)

**Comment by TheNorthMemory** [comment, marked as answer]
相对路径有可能找不到文件。比如， php程序运行时的的启动参数 `php /app/run.php`，配置文件的相对路径填的是 `'../certificates/wxpay_abcdef.pem'`，这得靠猜才能知道运行时的`..`相对位置是哪里。
